packages/integrations/src/offline.ts

```diff
@@ -0,0 +1,169 @@
+import { Event, EventProcessor, Hub, Integration } from '@sentry/types';
+import { getGlobalObject, logger, uuid4 } from '@sentry/utils';
+// @ts-ignore: Module '"localforage"' has no default export.
+import localforage from 'localforage';
+
+/**
+ * cache offline errors and send when connected
+ */
+export class Offline implements Integration {
+  /**
+   * @inheritDoc
+   */
+  public static id: string = 'Offline';
+
+  /**
+   * @inheritDoc
+   */
+  public readonly name: string = Offline.id;
+
+  /**
+   * the global instance
+   */
+  public global: Window;
+
+  /**
+   * the current hub instance
+   */
+  public hub?: Hub;
+
+  /**
+   * maximum number of events to store while offline
+   */
+  public maxStoredEvents: number;
+
+  /**
+   * event cache
+   */
+  public offlineEventStore: LocalForage; // type imported from localforage
+
+  /**
+   * @inheritDoc
+   */
+  public constructor(options: { maxStoredEvents?: number } = {}) {
+    this.global = getGlobalObject<Window>();
+    this.maxStoredEvents = options.maxStoredEvents || 30; // set a reasonable default
+    this.offlineEventStore = localforage.createInstance({
+      name: 'sentry/offlineEventStore',
+    });
+
+    if ('addEventListener' in this.global) {
```

Comment:
Event listener should be set up _after_ hub is set on instance (in `setupOnce` method) as event may be fired before hub is set (hub is used in `_sendEvents`).

IMHO it's bad practice to have business logic in constructor.

Reply:
Fixed in #2889